index.js

```diff
@@ -14,18 +17,64 @@ module.exports = {
   },
 
   setupPreprocessorRegistry: function(type, registry) {
-    if (type !== 'self') return;
+    this.setupJsPreprocessing(registry, type);
+
+    if (type === 'self') {
```

Comment:
that is correct, but for a different reason. we only look at the file tree in the addon to determine which components to generate or not